spec/classes/controller_spec.rb

```diff
@@ -4,10 +4,28 @@
   on_supported_os.each do |os, facts|
     context "on #{os}" do
       let(:facts) do
-        facts
+        facts.merge({:concat_basedir => '/tmp/puppetconcat'})
       end
 
-      it { should compile }
+      describe 'compiles when controller_addresses and address are specified' do
```

Comment:
Leave out the "compiles" here. That belongs to the `it { should compile }` below. The `describe` text should only contain the circumstances.